target/linux/ipq806x/files-5.4/arch/arm/boot/dts/qcom-ipq8065-rt4230w.dts

```diff
@@ -0,0 +1,373 @@
+						read-only;
+					};
+					ART: ART@1200000 {
+						label = "ART";
+						reg = <0x1200000 0x0140000>;
+						read-only;
+					};
+					BOOTCONFIG@0 {
+						label = "BOOTCONFIG";
+						reg = <0x1340000 0x0060000>;
+						read-only;
+					};
+					SBL2_1@0 {
+						label = "SBL2_1";
+						reg = <0x13a0000 0x0140000>;
+						read-only;
+					};
+					SBL3_1@0 {
+						label = "SBL3_1";
+						reg = <0x14e0000 0x0280000>;
+						read-only;
+					};
+					DDRCONFIG_1@0 {
+						label = "DDRCONFIG_1";
+						reg = <0x1760000 0x0120000>;
+						read-only;
+					};
+					SSD_1@0 {
+						label = "SSD_1";
+						reg = <0x1880000 0x0120000>;
+						read-only;
+					};
+					TZ_1@0 {
+						label = "TZ_1";
+						reg = <0x19a0000 0x0280000>;
+						read-only;
+					};
+					RPM_1@0 {
+						label = "RPM_1";
+						reg = <0x1c20000 0x0280000>;
+						read-only;
+					};
+					BOOTCONFIG1@0 {
+						label = "BOOTCONFIG1";
+						reg = <0x1ea0000 0x0060000>;
+						read-only;
+					};
+					APPSBL@0 {
+						label = "APPSBL";
+						reg = <0x1f00000 0x0500000>;
+						read-only;
+					};
+					ubi@0 {
+						label = "ubi";
+						reg = <0x2400000 0x1A000000>;
+					};
+				};
+			};
+		};
+
+		mdio0: mdio {
+			compatible = "virtual,mdio-gpio";
+			#address-cells = <1>;
+			#size-cells = <0>;
+			gpios = <&qcom_pinmux 1 GPIO_ACTIVE_HIGH>,
+				<&qcom_pinmux 0 GPIO_ACTIVE_HIGH>;
+			pinctrl-0 = <&mdio0_pins>;
+			pinctrl-names = "default";
+
+			phy0: ethernet-phy@0 {
+				device_type = "ethernet-phy";
+				reg = <0>;
+				qca,ar8327-initvals = <0x04 0x7600000 0x08 0x1000000 0x0c 0x80 0xe4 0xaa545 0xe0 0xc74164de 0x7c 0x4e 0x94 0x4e 0x50 0xcf02cf02 0x54 0xc832c832>;
+			};
+
+			phy4: ethernet-phy@4 {
+				device_type = "ethernet-phy";
+				reg = <4>;
+			};
+		};
+
+		gmac0: ethernet@37000000 {
+			status = "okay";
+			phy-mode = "rgmii";
+			qcom,id = <0>;
+
+			pinctrl-0 = <&rgmii2_pins>;
+			pinctrl-names = "default";
+
+			mtd-mac-address = <&ART 6>;
+
+			fixed-link {
+				speed = <1000>;
+				full-duplex;
+			};
+		};
+
+		gmac1: ethernet@37200000 {
+			status = "okay";
+			phy-mode = "sgmii";
+			qcom,id = <1>;
+
+			mtd-mac-address = <&ART 0>;
+
+			fixed-link {
+				speed = <1000>;
+				full-duplex;
+			};
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+		pinctrl-0 = <&button_pins>;
+		pinctrl-names = "default";
+
+		reset {
+			label = "reset";
+			gpios = <&qcom_pinmux 54 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+
+		wps {
+			label = "wps";
+			gpios = <&qcom_pinmux 68 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+		pinctrl-0 = <&led_pins>;
+		pinctrl-names = "default";
+		ledctrl1: ledctrl1 {
+			label = "rt4230w:ledctrl:1";
+			gpios = <&qcom_pinmux 22 GPIO_ACTIVE_HIGH>;
+		};
+
+		ledctrl2: ledctrl2 {
+			label = "rt4230w:ledctrl:2";
+			gpios = <&qcom_pinmux 23 GPIO_ACTIVE_HIGH>;
+		};
+
+		ledctrl3: ledctrl3 {
+			label = "rt4230w:ledctrl:3";
```

Comment:
Typically, LED labels are set up to a different logic. Maybe have look at some other DTS files here ...

Reply:
I set up the leds like this because based on how you change the 3 gpio pins, it does different things. For example:
on,off,off = red glow
on,off,on = red/blue glow
off,on,on = solid blue
etc.
A script needs to be made to control the led properly, unless there's a way to define multiple gpio pins under one led.